frontend/util.lua

```diff
@@ -400,4 +400,102 @@ function util.splitToArray(str, splitter, capture_empty_entity)
     return result
 end
 
+--- Convert a Unicode codepoint (number) to UTF8 char
+--
+--- @int c Unicode codepoint
+--- @treturn string UTF8 char
+function util.unicodeCodepointToUtf8(c)
+    if c < 128 then
+        return string.char(c)
+    elseif c < 2048 then
+        return string.char(192 + c/64, 128 + c%64)
+    elseif c < 55296 or 57343 < c and c < 65536 then
+        return string.char(224 + c/4096, 128 + c/64%64, 128 + c%64)
+    elseif c < 1114112 then
+        return string.char(240 + c/262144, 128 + c/4096%64, 128 + c/64%64, 128 + c%64)
+    else
+        return util.unicodeCodepointToUtf8(65533) -- U+FFFD REPLACEMENT CHARACTER
+    end
+end
+
+local HTML_ENTITIES_TO_UTF8 = {
+    ["&lt;"] = "<",
+    ["&gt;"] = ">",
+    ["&amp;"] = "&",
+    ["&quot;"] = '"',
+    ["&apos;"] = "'",
+    ["&nbsp;"] = "\xC2\xA0",
+    ["&#(%d+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x) ) end,
+    ["&#x(%x+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x,16) ) end,
```

Comment:
OK, I must say I'm not that proud about this util stuff. I just did enough so that the descriptions in the dozen of epub with descriptions I have would ALL display nice.
But I guess if we put that in util, it should be complete and not just enough for descriptions of a dozen books :|

I'll have a go with following your suggestions and pointers.
Your stackoverflow link lists many entities, most indeed used in french. In my dozen french book, I did not find a single such named entities (like ``&eacute;``). I don't know if I should double the size of our ``util.lua`` with these, or just add comments about this function limitations.

Also, this comprehensive list lists mostly latin named entities. Dunno if there are other ones in other alphabets/languages (I think there should be, greek probably have ``&alpha;``, ``&beta;`` ...).
Should I add such a big list of named entities in util.lua?